### PR TITLE
Bump Gradle, AGP and target/compileSdkVersion

### PR DESCRIPTION
**Pull request by rciovati**
I wanted to look into #75 when I realized I cannot import/build the project with a recent AS version, so here some housekeeping:
- Bump Gradle to 5.6.2
- Bump AGP to 3.5.0
- ~Bump compile/targetSdkVersion to 28. My assumption is that bumping to 29 will break the library (see #75) so for now I'd leave 28. On top of that, the latest support lib is 28.x.x so I'd rather move to AndroidX before changing that. Happy to do it in a separate PR.~
- As per @JakeWharton's comments it also bump compile/targetSdkVersion to 29

### CODE REVIEW

File: telescope/src/main/java/com/mattprecious/telescope/TelescopeLayout.java [hunk 1]
```diff
@@ -627,6 +629,7 @@ static Handler getBackgroundHandler() {
         final int width = displayMetrics.widthPixels;
         final int height = displayMetrics.heightPixels;
 
+        @SuppressLint("WrongConstant")
```
Comment (rciovati) [inline telescope/src/main/java/com/mattprecious/telescope/TelescopeLayout.java]: Lint complains for `PixelFormat.RGBA_8888` not being a valid constant here. Not sure what would be the appropriate replacement and how it worked so far.

File: telescope/src/main/java/com/mattprecious/telescope/TelescopeLayout.java [hunk 2]
```diff
@@ -410,9 +411,7 @@ private void cancel() {
   void trigger() {
     stop();
 
-    if (vibrate && hasVibratePermission(getContext())) {
-      vibrator.vibrate(VIBRATION_DURATION_MS);
-    }
+    vibrateIfNecessary();
```
Comment (rciovati) [inline telescope/src/main/java/com/mattprecious/telescope/TelescopeLayout.java]: Extracted to limit the scope of `@SuppressLint`

File: telescope/src/main/AndroidManifest.xml [hunk 3]
```diff
@@ -4,8 +4,6 @@
     android:versionCode="1"
     >
 
-  <uses-sdk android:minSdkVersion="14"/>
```
Comment (rciovati) [inline telescope/src/main/AndroidManifest.xml]: Gradle generates this now

**Comment by JakeWharton** [comment]
Changing to 29 won't affect the library or its users directly so there's no reason to hold off on it.

**Comment by rciovati** [comment]
@JakeWharton right, it won't break the library or its consumers. The sample app now has `targetSdkVersion=29` and it breaks but I guess it's a good start to actually fix the issue.

**Pull request merged**